src/metadata.c

```diff
@@ -15,45 +15,76 @@
 
 static const char *METADATA_STR = "METADATA";
 static const int METADATA_VERSION = 1;
-static const int METADATA_ELEM_COUNT = 4;
+static const int METADATA_ELEM_COUNT = 6;
 
 static char *metadataFilename(char *buf, size_t size, const char *filename)
 {
     safesnprintf(buf, size, "%s.meta", filename);
     return buf;
 }
 
-int MetadataWrite(Metadata *m, const char *filename, raft_term_t term,
-                  raft_node_id_t vote)
+void MetadataInit(Metadata *m)
 {
-    char buf[2048], orig[PATH_MAX], tmp[PATH_MAX];
-    int off = 0;
+    *m = (Metadata){
+        .node_id = RAFT_NODE_ID_NONE,
+        .dbid = "",
+        .term = 0,
+        .vote = RAFT_NODE_ID_NONE,
+    };
+}
+
+void MetadataTerm(Metadata *m)
+{
+    RedisModule_Free(m->filename);
+}
+
+void MetadataConfigure(Metadata *m, const char *filename, char *dbid,
+                       raft_node_id_t node_id)
+{
+    char buf[PATH_MAX];
+
+    RedisModule_Free(m->filename);
+    metadataFilename(buf, sizeof(buf), filename);
```

Comment:
Why not do it in `MetadataInit()`?

Reply:
Until we try to read metadata file, we don't know if the cluster is initialized. Currently, uninitialized RedisRaft allows reconfiguration of the `log_filename`. If we set filename in `MetadataInit()` and the cluster is not initialized, and then if admin changes the `log_filename`, metadata will still have the old name. 

So, I added another function, `MetadataConfigure()` which sets all the config(dbid, node_id, filename), when we know cluster is initialized: either on cluster init or on join. After that point, config cannot be changed anyway. 

The whole point is to allow admin to start an empty RedisRaft and be able to configure it from cli.